File: src/Api/Model/DB/Empresa.cs
using CriadorBaseDados.Model.DB.CNAE;
using Newtonsoft.Json;
using Realms;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;

namespace CriadorBaseDados.Model.DB
{
    [JsonObject(MemberSerialization.OptIn)]
    public class Empresa : RealmObject
    {
        [JsonProperty]
        [PrimaryKey]
        public string CNPJ { get; set; }
        [JsonProperty(DefaultValueHandling = DefaultValueHandling.Ignore)]
        public bool IsMatriz { get; set; }
        [JsonProperty]
        public string RazaoSocial { get; set; }
        [DefaultValue("")]
        [JsonProperty(DefaultValueHandling = DefaultValueHandling.Ignore)]
        public string NomeFantasia { get; set; }
        [Ignored]
        [JsonProperty]
        public int CodSituacao { get => Situacao.Cod; }
        public SituacaoCadastral Situacao { get; set; }
        [Ignored]
        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
        public int? CodMotivo { get => Motivo?.Cod; }
        public MotivoSituacao Motivo { get; set; }
        [JsonProperty]
        public int DataSituacao { get; set; }
        [Ignored]
        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
        public int? CodNaturezaJuridica { get => NaturezaJuridica?.Cod; }
        public NaturezaJuridica NaturezaJuridica { get; set; }
        [JsonProperty]
        public int DataInicioAtividade { get; set; }
        [Ignored]
        [JsonProperty]
        public int CodCNAE_Fiscal { get => CNAE_Fiscal.ID; }
        public CnaeSubclasse CNAE_Fiscal { get; set; }
        [JsonProperty]
        public Endereco Endereco { get; set; }
        [Ignored]
        [JsonProperty]
        public int CodMunicipio { get => Municipio.ID; }
        public Municipio Municipio { get; set; }
        [JsonProperty]
        public Contato Contato { get; set; }
        [Ignored]
        [JsonProperty]
        public int CodPorte { get => Porte.Cod; }
        public PorteEmpresa Porte { get; set; }
        [JsonProperty(DefaultValueHandling = DefaultValueHandling.Ignore)]
        public double CapitalSocial { get; set; }
        //[JsonProperty]
        [Backlink(nameof(SocioEmpresa.Empresa))]
        public IQueryable<SocioEmpresa> SociosEmpresa { get;}


    }
}

File: src/Mobile/b2b/lib/model/core/empresa.dart
import 'contato.dart';
import 'endereco.dart';
import 'socio_empresa.dart';

class Empresa {
  final String cnpj;
  final bool isMatriz;
  final String razaoSocial;
  final String nomeFantasia;
  final int codSituacao;
  final int? codMotivo;
  final int dataSituacao;
  final int? codNaturezaJuridica;
  final int dataInicioAtividade;
  final int codCnaeFiscal;
  final Endereco endereco;
  final int codMunicipio;
  final Contato? contato;
  final int codPorte;
  final double capitalSocial;
  final List<SocioEmpresa> sociosEmpresa;

  Empresa(
      this.cnpj,
      this.isMatriz,
      this.razaoSocial,
      this.nomeFantasia,
      this.codSituacao,
      this.codMotivo,
      this.dataSituacao,
      this.codNaturezaJuridica,
      this.dataInicioAtividade,
      this.codCnaeFiscal,
      this.endereco,
      this.codMunicipio,
      this.contato,
      this.codPorte,
      this.capitalSocial,
      this.sociosEmpresa);

  Empresa.fromJson(Map<String, dynamic> json)
      : cnpj = json['cnpj'],
        isMatriz = json['isMatriz'] ?? false,
        razaoSocial = json['razaoSocial'],
        nomeFantasia = json['nomeFantasia'] ?? "",
        codSituacao = json['codSituacao'],
        codMotivo = json['codMotivo'],
        dataSituacao = json['dataSituacao'],
        codNaturezaJuridica = json['codNaturezaJuridica'],
        dataInicioAtividade = json['dataInicioAtividade'],
        codCnaeFiscal = json['codCnaeFiscal'],
        endereco = Endereco.fromJson(json['endereco']),
        codMunicipio = json['codMunicipio'],
        contato = json['contato'] == null ? null : Contato.fromJson(json['contato']),
        codPorte = json['codPorte'],
        capitalSocial = json['capitalSocial'] ?? 0,
        sociosEmpresa = json['sociosEmpresa'] ?? List.empty(growable: true);

  Map<String, dynamic> toMap() {
    var map = <String, dynamic>{
      'cnpj': cnpj,
      'isMatriz': isMatriz == true ? 1 : 0,
      'razaoSocial': razaoSocial,
      'nomeFantasia': nomeFantasia,
      'codSituacao': codSituacao,
      'codMotivo': codMotivo,
      'dataSituacao': dataSituacao,
      'codNaturezaJuridica': codNaturezaJuridica,
      'dataInicioAtividade': dataInicioAtividade,
      'codCnaeFiscal': codCnaeFiscal,
      'endereco': endereco,
      'codMunicipio': codMunicipio,
      'contato': contato,
      'codPorte': codPorte,
      'capitalSocial': capitalSocial,
      'sociosEmpresa': sociosEmpresa,
    };
    return map;
  }
}
